arrabiata/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+use strum::EnumCount as _;
+
+pub mod column_env;
+pub mod columns;
+pub mod constraints;
+pub mod interpreter;
+pub mod logup;
+pub mod poseidon_3_60_0_5_5_fp;
+pub mod poseidon_3_60_0_5_5_fq;
+pub mod proof;
+pub mod prover;
+pub mod verifier;
+pub mod witness;
+
+/// The maximum degree of the polynomial that can be represented by the
+/// polynomial-time function the library supports.
+pub const MAX_DEGREE: u64 = 5;
+
+/// The minimum SRS size required to use Nova, in base 2.
+/// Requiring at least 2^16 to perform 16bits range checks.
+pub const MIN_SRS_LOG2_SIZE: usize = 16;
+
+/// The number of rows the IVC circuit requires.
+// FIXME: that might change. We use a vertical layout for now.
+pub const IVC_CIRCUIT_SIZE: usize = 1 << 13;
+
+/// The maximum number of columns that can be used in the circuit.
+pub const NUMBER_OF_COLUMNS: usize = 17;
+
+/// The maximum number of public inputs the circuit can use per row
+/// We do have 12 for now as we want to compute 4 rounds of poseidon per row.
+/// In addition to the 12 public inputs required for the rounds, we add 2 more
+/// for the values to absorb.
+// FIXME: we can do 5 rounds per row by using "the next row"
+pub const NUMBER_OF_PUBLIC_INPUTS: usize = 12 + 2;
+
+/// The low-exponentiation value used by the Poseidon hash function for the
+/// substitution box.
+///
+/// The value is used to perform initialisation checks with the fields.
+pub const POSEIDON_ALPHA: u64 = 5;
```

Comment:
all the poseidon related constant should not be there
Maybe they can be given by the curves, since they come from there

Reply:
They do not come necessarily from the curves. They come from the design. 7 could also be used for instance.